lib/ansible/modules/packaging/os/snap.py

```diff
@@ -0,0 +1,193 @@
+    returned: When snaps are installed
+cmd:
+    description: The command that was executed on the host
+    type: string
+    returned: When changed is true
+'''
+
+from ansible.module_utils.basic import AnsibleModule
+import re
+
+
+def snap_exists(module, snap_name):
+    snap_path = module.get_bin_path("snap", True)
+    cmd_parts = [snap_path, 'info', snap_name]
+    cmd = ' '.join(cmd_parts)
+    rc, out, err = module.run_command(cmd, check_rc=False)
+
+    return rc == 0
+
+
+def is_snap_installed(module, snap_name):
+    snap_path = module.get_bin_path("snap", True)
+    cmd_parts = [snap_path, 'list', snap_name]
+    cmd = ' '.join(cmd_parts)
+    rc, out, err = module.run_command(cmd, check_rc=False)
+
+    return rc == 0
+
+
+def install_snaps(module, snap_names):
+    snaps_not_installed = list()
+    for snap_name in snap_names:
+        if is_snap_installed(module, snap_name):
+            continue
+        snaps_not_installed.append(snap_name)
+
+    if not snaps_not_installed:
+        module.exit_json(classic=module.params['classic'], changed=False)
+
+    if module.check_mode:
+        module.exit_json(classic=module.params['classic'], changed=True)
+
+    classic = '--classic' if module.params['classic'] else ''
+
+    snap_path = module.get_bin_path("snap", True)
+    cmd_parts = [snap_path, 'install'] + snaps_not_installed + [classic]
+    cmd = ' '.join(cmd_parts)
+
+    # Actually install the snaps
+    rc, out, err = module.run_command(cmd, check_rc=False)
+
+    if rc == 0:
+        module.exit_json(classic=module.params['classic'], changed=True, cmd=cmd, stdout=out, stderr=err)
+    else:
+        m = re.match(r'^error: This revision of snap "(?P<package_name>\w+)" was published using classic confinement', err)
+        if m is not None:
+            err_pkg = m.group('package_name')
+            msg = "Couldn't install {name} because it requires classic confinement".format(name=err_pkg)
+        else:
+            # The error is not related to the confinement
+            msg = "Something went wrong"
+        module.fail_json(msg=msg, classic=module.params['classic'], cmd=cmd, stdout=out, stderr=err)
+
+
+def remove_snaps(module, snap_names):
+    snaps_installed = list()
+    for snap_name in snap_names:
+        if not is_snap_installed(module, snap_name):
+            continue
+        snaps_installed.append(snap_name)
+    if not snaps_installed:
+        module.exit_json(changed=False)
+
+    if module.check_mode:
+        module.exit_json(changed=True)
+
+    snap_path = module.get_bin_path("snap", True)
+    cmd_parts = [snap_path, 'remove'] + snaps_installed
+    cmd = ' '.join(cmd_parts)
+
+    # Actually remove the snaps
+    rc, out, err = module.run_command(cmd, check_rc=False)
+
+    if rc == 0:
+        module.exit_json(changed=True, cmd=cmd, stdout=out, stderr=err)
+    else:
+        module.fail_json(msg="Something went wrong.", cmd=cmd, stdout=out, stderr=err)
+
+
+def main():
+    module_args = dict(
```

Comment:
Please turn dict creation into a literal (`{}`)

Reply:
Done 👍
Is there a reason for this?

Reply:
It's better readable, and faster